Mathlib/Combinatorics/SimpleGraph/Metric.lean

```diff
@@ -122,6 +122,30 @@ theorem dist_comm {u v : V} : G.dist u v = G.dist v u := by
     simp [h, h', dist_eq_zero_of_not_reachable]
 #align simple_graph.dist_comm SimpleGraph.dist_comm
 
+lemma Reachable.of_dist_ne_zero {u v : V} (h : G.dist u v ≠ 0) : G.Reachable u v := by
+  apply dist_eq_zero_iff_eq_or_not_reachable.not.mp at h
```

Comment:
Do we want a dist_ne_zero... lemma?

Reply:
This was recommended by @urkud [here](https://github.com/leanprover-community/mathlib4/pull/11945#discussion_r1569808127), I think it is a good idea because this lemma was useful in #12278.

Reply:
Sorry, I meant the iff version - a normalized version of `dist_eq_zero_iff_eq_or_not_reachable.not`

Reply:
Good idea, done!